lib/memoize.js

```diff
@@ -63,7 +66,11 @@ export default function memoize(fn, hasher) {
             queues[key] = [callback];
             _fn.apply(null, args.concat(function(/*args*/) {
                 var args = slice(arguments);
-                memo[key] = args;
+                var err = args[0];
+                // #1465 don't memoize if an error occurred
+                if (!err) {
+                    memo[key] = args;
+                }
                 var q = queues[key];
                 delete queues[key];
                 for (var i = 0, l = q.length; i < l; i++) {
```

Comment:
We should keep this as is for now, but for v3, I'd be open to changing this behaviour, as there's a potential race condition here (similar to #1248).

If the initial call to the memoized function takes between `m` and `n` seconds to err, and a second call is made `x` seconds after the initial (m < x < n`) then the function might only be run once depending on whether or not the initial call finished. To eliminate it, we would have to call the memoized function again if there are queued calls, and an error occurred.